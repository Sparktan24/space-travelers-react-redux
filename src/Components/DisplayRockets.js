/* eslint-disable arrow-body-style */
import { useSelector } from 'react-redux';

const DisplayRockets = () => {
  const { rockets } = useSelector((store) => store.rockets);
  const rocketFilter = rockets.filter((rocket) => rocket.reserved);

  return (
    <>
      {rocketFilter.length > 0 && (
        <table className="profile-table">
          <tbody>
            {rocketFilter.map((rocket) => {
              return rocket.reserved && (
                <tr key={rocket.id}>
                  <td>{rocket.name}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      )}
      {!rocketFilter.length && (<p className="profile-table">No rockets reserved</p>)}
    </>
  );
};

export default DisplayRockets;
