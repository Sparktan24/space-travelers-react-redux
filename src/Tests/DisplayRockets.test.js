import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { useSelector } from 'react-redux';
import DisplayRockets from '../Components/DisplayRockets';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('DisplayRcplets component', () => {
  it('Renders reserved rockets correctly', () => {
    const mockedRockets = [
      { id: 1, name: 'Falcon 1', reserved: true },
      { id: 2, name: 'Falcon 9', reserved: true },
      { id: 3, name: 'Falcon Heavy', reserved: false },
    ];

    useSelector.mockImplementation((selectorFn) => selectorFn(
      { rockets: { rockets: mockedRockets } },
    ));

    const { container, getByText } = render(<DisplayRockets />);

    expect(getByText('Falcon 1')).toBeInTheDocument();
    expect(getByText('Falcon 9')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders "No missions assigned" when no reserved missions', () => {
    useSelector.mockImplementation((selectorFn) => selectorFn({ rockets: { rockets: [] } }));

    const { container, getByText } = render(<DisplayRockets />);

    expect(getByText('No rockets reserved')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
