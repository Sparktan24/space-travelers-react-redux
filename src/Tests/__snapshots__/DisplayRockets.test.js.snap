// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayRcplets component Renders reserved rockets correctly 1`] = `
<table
  class="profile-table"
>
  <tbody>
    <tr>
      <td>
        Falcon 1
      </td>
    </tr>
    <tr>
      <td>
        Falcon 9
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`DisplayRcplets component renders "No missions assigned" when no reserved missions 1`] = `
<p
  class="profile-table"
>
  No rockets reserved
</p>
`;
